docs(releases): v1.17.0 release notes (#27225)
PX4/PX4-Autopilot: docs/en/releases/1.17.md

  </div>
</div>

PX4 v1.17 builds on [PX4 v1.16](../releases/1.16.md), with the changes below landing since v1.16.2.

> This release adds Altitude Cruise mode, improves Fixed Wing Takeoff behaviour on navigation loss, and exposes cleaner high-level fixed-wing and rover control interfaces for ROS 2 workflows.

Reviewer: Where possible cross link. Caveat, docs may not existing in v1.17 branch. So you need to decide what to backport.

```suggestion
This release adds [Altitude Cruise](../flight_modes_mc/altitude_cruise.md) mode, improves Fixed Wing Takeoff behaviour on navigation loss, and exposes cleaner high-level fixed-wing and rover control interfaces for ROS 2 workflows.
```

PX4 v1.17 also includes user-visible MAVLink, RC, logging, failsafe, and rover refinements across the stack.

::: warning
PX4 v1.17 is documented on an alpha/beta release branch.
See [PX4-Autopilot `main` Release Notes](../releases/main.md) for newer changes on `main`.
:::

## Read Before Upgrading

Please continue reading for [upgrade instructions](#upgrade-guide).

## Major Changes

- <Badge type="warning" text="Experimental" /> [MC Neural Network Control](../neural_networks/mc_neural_network_control.md): PX4 gains first-class on-device neural-network inference.
  With [TensorFlow Lite Micro](../neural_networks/tflm.md) now integrated into the firmware, you can take a network you trained externally (e.g. via reinforcement learning in [Aerial Gym](https://ntnu-arl.github.io/aerial_gym_simulator/)), convert it to tflite, and run it on the autopilot as a full replacement for the multicopter controller stack.

>   ([PX4-Autopilot#24366](https://github.com/PX4/PX4-Autopilot/pull/24366))

Reviewer: How do you know it is first class - what does that mean? What is a neural-network inference? 

IMO this is experimental so it is the last thing in the list.
I think you should position it as test code, whereas now it sounds like next revision of PX4 this will replace the flight stack.

- New multicopter flight mode: [Altitude Cruise](../flight_modes_mc/altitude_cruise.md).
  Holds tilt and heading on stick release so the vehicle keeps cruising at a steady velocity instead of stopping like Altitude mode does.
- [Fixed Wing Takeoff mode](../flight_modes_fw/takeoff.md) now keeps climbing with level wings on navigation loss and can use the takeoff waypoint latitude and longitude to define the loiter position.
  ([PX4-Autopilot#25226](https://github.com/PX4/PX4-Autopilot/pull/25226))
- Fixed-wing vehicles (and VTOLs in fixed-wing mode) can now be controlled from ROS 2 via the new [`FwLateralLongitudinalSetpointType`](../ros2/px4_ros2_control_interface.md#fw-lateral-longitudinal-setpoint) in the PX4 ROS 2 Control Interface, exposing direct lateral and longitudinal setpoints.
- Rovers can now be controlled from ROS 2 via the new [`RoverSetpointTypes`](../ros2/px4_ros2_control_interface.md#rover-setpoints) in the PX4 ROS 2 Control Interface, with valid combinations of position, speed, throttle, attitude, rate, and steering setpoints exposed as guaranteed-valid setpoint types. See [Rovers Apps & API](../flight_modes_rover/api.md).

> - <Badge type="warning" text="Experimental" /> The in-tree [Zenoh middleware](../middleware/zenoh.md) matures to `rmw_zenoh` compatibility (CDRv1 serialization, ROS 2 graph liveliness, auto-generated config from `dds_topics.yaml`, Domain ID parameter, Zenoh CLI). Zenoh is built into the default firmware on FMU-v6xRT (`make px4_fmu-v6xrt_default`); on FMU-v6x and SITL it ships as an opt-in `zenoh` build variant (`make px4_fmu-v6x_zenoh`, `make px4_sitl_zenoh`).

Reviewer: ```suggestion
- <Badge type="warning" text="Experimental" /> The in-tree [Zenoh middleware](../middleware/zenoh.md) matures to `rmw_zenoh` compatibility (CDRv1 serialization, ROS 2 graph liveliness, auto-generated config from `dds_topics.yaml`, Domain ID parameter, Zenoh CLI). Zenoh is built into the default firmware on FMU-v6xRT (`make px4_fmu-v6xrt_default`); on FMU-v6x and SITL it ships as a `zenoh` build variant (`make px4_fmu-v6x_zenoh`, `make px4_sitl_zenoh`).
```

> ## Upgrade Guide

Reviewer: In what way are the following an upgrade guide?

Reviewer: Ah, I see you just copied the pattern of the last two releases. 

This is an upgrade guide: https://docs.px4.io/v1.16/en/releases/1.14#upgrade-guide

The difference is that it explains what you need to do when upgrading, perhaps with a sprinkling of what you need to know. This is all the later and none of the former.

  ([PX4-Autopilot#25319](https://github.com/PX4/PX4-Autopilot/pull/25319), [PX4-Autopilot#25322](https://github.com/PX4/PX4-Autopilot/pull/25322))
- ARK SCH16T driver updated for new Murata modules.
  ([PX4-Autopilot#24029](https://github.com/PX4/PX4-Autopilot/pull/24029))
- mRO PixracerPro: enable additional UARTs.
  ([PX4-Autopilot#22516](https://github.com/PX4/PX4-Autopilot/pull/22516))
- 3DR Control Zero H7 OEM Rev G: MTD driver fix and missing module that prevented USB enumeration.
  ([PX4-Autopilot#25015](https://github.com/PX4/PX4-Autopilot/pull/25015))
- ESP32: PWM register updates correctly on change.
  ([PX4-Autopilot#25653](https://github.com/PX4/PX4-Autopilot/pull/25653))
- Tropic-Community baseboard (Teensy 4.1): PX4 config refresh, DTCM moved to heap with vectors in ITCM.

### Common

- [QGC Bootloader Update](../advanced_config/bootloader_update.md#qgc-bootloader-update-sys-bl-update) via the [`SYS_BL_UPDATE`](../advanced_config/parameter_reference.md#SYS_BL_UPDATE) parameter has been re-enabled after being broken for several releases.
  ([PX4-Autopilot#25032](https://github.com/PX4/PX4-Autopilot/pull/25032))

> - Failsafe handling now lets the pilot take over from a degraded failsafe state.

Reviewer: EDIT> Ignore. Just a bug.

- **NuttX support**: update NuttX config for use with Zenoh; backport NuttX FMU-v6x config for Zenoh ([PX4-Autopilot#26213](https://github.com/PX4/PX4-Autopilot/pull/26213)).
- **SITL**: new `px4_sitl_zenoh` cmake variant, default to `127.0.0.1` on SITL/POSIX, autostart Zenoh in SITL when enabled, friendlier error message when scouting finds nothing, GCC/SITL compile fixes.
- **Robustness**: validate payload size before stack allocation, increase CDR safety margin, handle parsing errors and non-existing types in config, fix topic-name and datatype mapping, pubsub-factory datatype-naming fix, fix status `keyexpr` printf, prevent integer-conversion warnings in zenoh-pico, optimize memory usage with optional publish-on-matching.
- **Docs and OOM fixes (backport)**: [PX4-Autopilot#26053](https://github.com/PX4/PX4-Autopilot/pull/26053).
- **Submodule**: `zenoh-pico` updated.

### MAVLink

- MAVFTP hardening: rejects path-traversal sequences in FTP operations and validates sessions in FTP write/burst.
- MAVLink v1 is now opt-in via [`MAV_PROTO_VER`](../advanced_config/parameter_reference.md#MAV_PROTO_VER).
  ([PX4-Autopilot#25583](https://github.com/PX4/PX4-Autopilot/pull/25583))

> - Extended `MISSION_CURRENT` message implemented.

Reviewer: No PR number. This means the opaque ids right?

- Added [Apps & API](../flight_modes_rover/api.md) support including [Rover Setpoints](../ros2/px4_ros2_control_interface.md#rover-setpoints).
- [Rover simulation](../frames_rover/index.md#simulation) updated to match the reworked rover architecture.
  ([PX4-Autopilot#25644](https://github.com/PX4/PX4-Autopilot/pull/25644))
- Continued per-vehicle (Ackermann/differential/mecanum) restructuring with separated actuator control, position-control updates, mode-management centralization, and module split into folders.
- New stick-input scaling parameters.
  ([PX4-Autopilot#25427](https://github.com/PX4/PX4-Autopilot/pull/25427))
- Improved rover hold-position logic.
  ([PX4-Autopilot#25466](https://github.com/PX4/PX4-Autopilot/pull/25466))
- `RX_MAX_THR_YAW_R` replaced by `RO_YAW_RATE_CORR`.
- Manual yaw-rate input: exponential scaling.
- Comprehensive [Rover API documentation](../flight_modes_rover/api.md).
  ([PX4-Autopilot#25499](https://github.com/PX4/PX4-Autopilot/pull/25499))
- MR-CANHUBK344 NXP B3RB rover platform.
  ([PX4-Autopilot#23897](https://github.com/PX4/PX4-Autopilot/pull/23897))
- Aion Robotics R1 Rover airframe (`50001`) marked discontinued.

### Safety / Commander

> - Pilot can take over from a degraded failsafe action.

Reviewer: Cool. Note is repeat of earlier item
